core/src/io/Dataparser.cpp

```diff
@@ -1059,32 +1070,31 @@ namespace IO
             myfile.Read_String( ovf_data_representation, "# Begin: Data" );
             std::istringstream repr( ovf_data_representation );
             repr >> ovf_data_representation;
-            if( ovf_data_representation == "Binary" ) 
+            if( ovf_data_representation == "binary" ) 

```

Comment:
I believe this is incorrect, as the `ovf_binary_length` is not read in via the `Filter_File_Handle::Read_String` function. As it is therefore not lowercased, this will nor compare correctly.
An alternative that would be more readable in code would be to write a function `IO::lowercase(std::string & str)`, the comparison could then look like `if( lowercase(ovf_data_representation) == "binary")`.

Reply:
The `ovf_data_representation` is read, together with the keyword `"Begin: Data"`, by the `Read_String()` method of `Filter_File_Handle` object. At first the line under investigation is de-capitalized by `Filter_File_Handle::GetLine()` function. Additionally, the `keyword` argument is de-capitalized by `Read_String()` to assure that the keyword will matched in the line no matter how the `keyword` was formatted in the call of the function. For that reason the `ovf_data_representation` variable can only be lower case. 

If we change in the code the comparison string `"binary"` to `"Binary"` the `Read_From_OVF()` will fail. It would make sense to do the inverse of what you are proposing eg `ovf_data_representation == lowercase("Binary")` but of course one can just simply write `"binary"` instead.